File: luna-monitor/src/luna_monitor/ui/colors.py
"""Color threshold functions for system metrics."""


def pct_color(pct: float) -> str:
    """Color by percentage: cyan < 60 < yellow < 85 < red."""
    if pct >= 85:
        return "red"
    if pct >= 60:
        return "yellow"
    return "cyan"


def temp_color(celsius: float) -> str:
    """Color by temperature: green < 70 < yellow < 85 < red."""
    if celsius >= 85:
        return "red"
    if celsius >= 70:
        return "yellow"
    return "green"


def io_color(speed_bps: float) -> str:
    """Color I/O speed by intensity: dim < 1MB < cyan < 10MB < yellow < 100MB < red."""
    mb = speed_bps / 1e6
    if mb >= 100:
        return "red bold"
    if mb >= 10:
        return "yellow bold"
    if mb >= 1:
        return "cyan"
    return "dim"


# Claude panels use distinct colors to separate from system panels
CLAUDE_BORDER = "cyan"
SYSTEM_BORDER = "bright_black"
BURNDOWN_STYLE = "magenta"
CPU_STYLE = "cyan"

File: luna-monitor/crates/luna-monitor/src/ui/colors.rs
use ratatui::style::Color;

pub const CLAUDE_BORDER: Color = Color::Cyan;
pub const SYSTEM_BORDER: Color = Color::DarkGray;
pub const BURNDOWN_COLOR: Color = Color::Magenta;
pub const CPU_COLOR: Color = Color::Cyan;

pub fn pct_color(pct: f64) -> Color {
    if pct < 60.0 {
        Color::Cyan
    } else if pct < 85.0 {
        Color::Yellow
    } else {
        Color::Red
    }
}

pub fn temp_color(c: f64) -> Color {
    if c < 70.0 {
        Color::Green
    } else if c < 85.0 {
        Color::Yellow
    } else {
        Color::Red
    }
}

pub fn io_color(bps: f64) -> Color {
    let mb = bps / (1024.0 * 1024.0);
    if mb < 1.0 {
        Color::DarkGray
    } else if mb < 10.0 {
        Color::Cyan
    } else if mb < 100.0 {
        Color::Yellow
    } else {
        Color::Red
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_pct_color_ranges() {
        assert_eq!(pct_color(30.0), Color::Cyan);
        assert_eq!(pct_color(59.9), Color::Cyan);
        assert_eq!(pct_color(60.0), Color::Yellow);
        assert_eq!(pct_color(84.9), Color::Yellow);
        assert_eq!(pct_color(85.0), Color::Red);
        assert_eq!(pct_color(100.0), Color::Red);
    }

    #[test]
    fn test_temp_color_ranges() {
        assert_eq!(temp_color(50.0), Color::Green);
        assert_eq!(temp_color(75.0), Color::Yellow);
        assert_eq!(temp_color(90.0), Color::Red);
    }

    #[test]
    fn test_io_color_ranges() {
        assert_eq!(io_color(0.0), Color::DarkGray);
        assert_eq!(io_color(5.0 * 1024.0 * 1024.0), Color::Cyan);
        assert_eq!(io_color(50.0 * 1024.0 * 1024.0), Color::Yellow);
        assert_eq!(io_color(200.0 * 1024.0 * 1024.0), Color::Red);
    }
}
